RileyLinkBLEKit/RileyLinkBLEDevice.m

```diff
@@ -464,7 +464,7 @@ - (void) setCustomName:(nonnull NSString*)customName {
 }
 
 - (void) onIdle {
-    if (idleListeningEnabled && _peripheral.state == CBPeripheralStateConnected) {
+    if (!runningIdle && idleListeningEnabled && _peripheral.state == CBPeripheralStateConnected && dataCharacteristic != nil) {
```

Comment:
The problem with requiring `!runningIdle` here is that it's sometimes never reset to `NO` (like if the RL goes out-of-range before the idle timeout). It's the explicit reason why line 495 is an OR:
```objc
!runningIdle || ([NSDate dateWithTimeIntervalSinceNow:-resetIdleAfterInterval] > _lastIdle
```
The second condition is met when the system "misses" the timeout.